mage_ai/streaming/sources/mongodb.py

```diff
@@ -0,0 +1,69 @@
+from dataclasses import dataclass
+from typing import Any, Callable, Mapping, Optional
+
+from bson.timestamp import Timestamp
+from pymongo import MongoClient
+from pymongo.typings import _Pipeline
+
+from mage_ai.shared.config import BaseConfig
+from mage_ai.streaming.sources.base import BaseSource
+
+
+@dataclass
+class MongoDBConfig(BaseConfig):
+    connection_str: str
+    database: str
+    collection: str = None
+    batch_size: Optional[int] = 100
+    pipeline: Optional[_Pipeline] = None
+    operation_time: Optional[Timestamp] = None
+    start_after: Optional[Mapping[str, Any]] = None
+
+
+class MongoSource(BaseSource):
+    config_class = MongoDBConfig
+
+    def init_client(self):
+        connection_str = self.config.connection_str
+        database = self.config.database
+        collection = self.config.collection
+        batch_size = self.config.batch_size
+        pipeline = self.config.pipeline
+        operation_time = self.config.operation_time
+        start_after = self.config.start_after
+        self._print(f"Starting to MongoDB Stream for {database}")
+        self._print(f"You can use this collection {collection}")
+        self._print(f"You can use {connection_str}")
+        self._print(f"You can use {batch_size}")
+        self._print(f"You can use {pipeline}")
+        self._print(f"You can use {operation_time}")
+        self._print(f"You can use {start_after}")
+
+    def read(self, handler: Callable):
+        pass
+
+    def batch_read(self, handler: Callable):
+        self._print("Start getting message for MongoDB streaming.")
+        try:
+            client = MongoClient(self.config.connection_str)
```

Comment:
We usually initialize the client in the `init_client` method.

Reply:
I don't understand that this problem. I write a test to see another streaming but this test is using init_client. Can you explain clearly please?

Reply:
I mean this `self.client = MongoClient(self.config.connection_str)` code should be put into the `init_client` method.
You can check other sources for example (e.g. https://github.com/mage-ai/mage-ai/blob/master/mage_ai/streaming/sources/azure_event_hub.py#L19)

Reply:
But i use db variable in client side. 

```
client = MongoClient(self.config.connection_str)
db = client.get_database(self.config.database)

```

Maybe I write

```
 db = self.client.get_database(self.config.database)

```

Reply:
yes